python/paddle/hapi/model_summary.py

```diff
@@ -163,7 +165,20 @@ def _check_input(input_size):
             return [_check_input(i) for i in input_size]
 
     _input_size = _check_input(_input_size)
-    result, params_info = summary_string(net, _input_size, dtypes)
+
+    if input is None:
+        _input = input
+    else:
+        if isinstance(input, dict):
+            _input = []
+            for item in input.keys():
+                _input.append(input[item])
+        elif isinstance(input, list):
+            _input = input
+        else:
+            raise Exception('Input must be list or dict.')
```

Comment:
input can be any type as long as the network can run through

Reply:
done